pkg/proctree/exit_processing.go

```diff
@@ -0,0 +1,96 @@
+package proctree
+
+import (
+	"fmt"
+
+	"github.com/aquasecurity/tracee/signatures/helpers"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// processExitEvent remove references of processes from the tree when the corresponding process exit without children, or
+// if the last child process of a process exits.
+// Notice that there is a danger of memory leak if there are lost events of sched_process_exit (but is limited to the
+// possible number of PIDs - 32768)
+func (tree *ProcessTree) processExitEvent(event *trace.Event) error {
+	err := tree.processDefaultEvent(event)
+	if err != nil {
+		return err
+	}
+	process, err := tree.getProcess(event.HostProcessID)
+	if err != nil {
+		return fmt.Errorf("process was inserted to the treee but is missing right after")
+	}
+	process.Mutex.Lock()
+	defer process.Mutex.Unlock()
+	thread, ok := process.Threads.Get(event.HostThreadID)
+	if !ok {
+		thread = &threadInfo{}
+	}
+	thread.exitTime = timestamp(event.Timestamp)
+	process.Threads.Set(event.HostThreadID, thread)
+
+	processGroupExit, err := helpers.GetTraceeBoolArgumentByName(*event, "process_group_exit")
+	if err != nil {
+		return err
+	}
+
+	if processGroupExit {
+		process.IsAlive = false
+		process.ExitTime = timestamp(event.Timestamp)
+		for _, tid := range process.Threads.Keys() {
+			info, ok := process.Threads.Get(tid)
+			if ok && info.exitTime == 0 {
+				info.exitTime = timestamp(event.Timestamp)
+			}
+		}
+		tree.cachedDeleteProcess(process.InHostIDs.Pid)
+	}
+	return nil
+}
+
+const cachedDeadEvents = 100
```

Comment:
Why did you chose this value specifically?

Reply:
So I have no reason, just an hunch.
The whole size of the cache is a problematic issue, as we can't predict the load.
The number here is just me trying to put a limit to the amount of memory used for dead nodes.